### PR TITLE
Fix fate debug message. Closes #2838

### LINKED ISSUE #2838: Fate debug message is missing cause
While looking at #2837 I found that AcceptableException.getCause() is always empty. [This](https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/fate/Fate.java#L186) message in the manager log is always missing the cause. Looking at the class hierarchy of `AcceptableThriftTableOperationException`, I don't think that `cause` is ever set.

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

**Review by dlmarion — APPROVED** [review]
This might work, I remember that the cause was empty. Did you happen to test this to see what  the message is for the parent exception?

**Comment by milleruntime** [comment]
> This might work, I remember that the cause was empty. Did you happen to test this to see what the message is for the parent exception?

No, I wanted to ask what you did to see the error...

**Comment by dlmarion** [comment]
[This](https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/fate/Fate.java#L186) message is in the log, but without the stack trace

**Comment by dlmarion** [comment]
Run `ConcurrentDeleteTableIT.testConcurrentDeleteTablesOps`, then look in the MAC Manager out file

**Comment by milleruntime** [comment]
OK. It looks like the message is null too. Here is the full stacktrace:
<pre>
2022-08-03T15:29:31,308 [fate.Fate] DEBUG: Failed to execute Repo, FATE[14ccaf316c860ecd]
org.apache.accumulo.core.clientImpl.AcceptableThriftTableOperationException: null
        at org.apache.accumulo.manager.tableOps.Utils.reserveTable(Utils.java:96) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.delete.DeleteTable.isReady(DeleteTable.java:45) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.delete.DeleteTable.isReady(DeleteTable.java:30) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.TraceRepo.isReady(TraceRepo.java:47) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.fate.Fate$TransactionRunner.run(Fate.java:89) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
</pre>

Here is the code throwing the exception:
<pre>
throw new AcceptableThriftTableOperationException(tableId.canonical(), "", op,
              TableOperationExceptionType.NOTFOUND, "Table does not exist");
</pre>

So there is a lot of information that is being lost over the wire.

**Comment by dlmarion** [comment]
I think the issues is that there are no ThriftTableOperationException constructors that take a `cause` or `message`. The constructors on the Thrift objects don't call the super class constructors to set the variables. It looks like ThriftTableOperationException doesn't consider TException at all in toString, equals, and hashCode.

Looks like it's a known issue for about 7 years: https://issues.apache.org/jira/browse/THRIFT-3173

File: core/src/main/java/org/apache/accumulo/core/clientImpl/AcceptableThriftTableOperationException.java [hunk 1]
```diff
@@ -30,9 +30,44 @@ public class AcceptableThriftTableOperationException extends ThriftTableOperatio
...
+  }
+
+  @Override
+  public TableOperationExceptionType getType() {
+    return type;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
```
Comment (dlmarion) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/AcceptableThriftTableOperationException.java:71]: I'm not sure you need to do all of this, it's in the parent class, right?

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/AcceptableThriftTableOperationException.java]: Yeah, I dropped them.

**Comment by milleruntime** [comment]
OK I added the table ID, name and description to the log statement. They could also be null but if the information is there it will get printed and that is better than nothing.
<pre>
2022-08-04T11:03:23,829 [fate.Fate] DEBUG: Failed to execute Repo FATE[0b4d7e1ccdd8dd1c] for (1) Table does not exist
</pre>

**Comment by ctubbsii** [comment]
Reminder: please set the Project version(s) when merging PRs, so we can track the fix version for changes.

**Comment by milleruntime** [comment]
> Reminder: please set the Project version(s) when merging PRs, so we can track the fix version for changes.

FYI the original ticket had the version.

**Pull request merged**